### PR TITLE
s3 storage for spark-history server

### PR DESCRIPTION
**Pull request by ntfrnzn**
Feat: create s3 storage for spark-history server on initialization

Introduces a cloudstorage interface which is faked and mocked for tests,
and allows for different implementatons on different cloud providers.

https://spotinst.atlassian.net/browse/BGD-373

### CODE REVIEW

File: install/installer.go [hunk 1]
```diff
@@ -275,13 +291,16 @@ func (i *HelmInstaller) Delete(chartName string, repository string, version stri
...
-	cfg, err := i.getActionConfig(catalog.SystemNamespace)
+	cfg, err := i.getActionConfig(i.namespace)
 	if err != nil {
 		return fmt.Errorf("failed to get action config, %w", err)
 	}
 
 	getAction := action.NewUninstall(cfg)
 	rel, err := getAction.Run(releaseName)
+	if err != nil {
+		i.Log.Error(err, "ignoring deletion error sorry")
```
Comment (thorsteinnth) [inline install/installer.go:302]: maybe remove "sorry" :)

File: internal/aws/s3.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	if err := t.Execute(&output, bucket); err != nil {
+		return "", err
+	}
+
+	return output.String(), nil
+}
+
+func getRegionFromMetadata() (string, error) {
+	// curl http://169.254.169.254/latest/dynamic/instance-identity/document
+	r, err := http.Get("http://169.254.169.254/latest/dynamic/instance-identity/document")
```
Comment (thorsteinnth) [inline internal/aws/s3.go:143]: I guess this messes with local development :(

**Review by thorsteinnth — APPROVED** [review]

File: install/installer.go (hunk 1 above)
Comment (ntfrnzn) [reply install/installer.go:302]: good point.

File: internal/aws/s3.go (hunk 2 above)
Comment (ntfrnzn) [reply internal/aws/s3.go:143]: the method is only called if the config.Region is empty -- in local development it should have been populated.

**Pull request merged**